searx/poolrequests.py

```diff
@@ -111,6 +111,32 @@ def get_time_for_thread():
     return threadLocal.total_time
 
 
+def get_proxy_cycles(proxy_settings):
```

Comment:
Maybe add a few unit tests? These functions can be tested quite easily.

Reply:
See last commit, tests of:
* get_proxy_cycles : old syntax, new syntax.
* get_proxies
* request: with and without the `proxies=` parameter